Bonsai.Design/ObjectTextVisualizer.cs

```diff
@@ -30,27 +31,90 @@ public class ObjectTextVisualizer : BufferedVisualizer
+                sb.Clear();
+                sb.EnsureCapacity(rawText.Length);
+                MakeDisplayString(rawText, sb);
+                DisplayString(sb.ToString());
             }
+
+            // Update the visual representation of the buffer
+            Debug.Assert(buffer.Count <= bufferSize);
+            sb.Clear();
+            foreach (var line in buffer)
+            {
+                if (sb.Length > 0)
+                    sb.Append(Environment.NewLine);
+                sb.Append(line);
+            }
+            textBox.Text = sb.ToString();
+            textPanel.Invalidate();
         }
 
         /// <inheritdoc/>
         public override void Show(object value)
         {
-            value ??= string.Empty;
-            var text = value.ToString();
-            text = Regex.Replace(text, @"\r|\n", string.Empty);
-            buffer.Enqueue(text);
-            while (buffer.Count > bufferSize)
+            var rawText = value?.ToString() ?? string.Empty;
+            var stringBuilder = new StringBuilder(rawText.Length);
+            MakeDisplayString(rawText, stringBuilder);
+            DisplayString(stringBuilder.ToString());
+        }
+
+        /// <summary>Converts the specified string for display in this visualizer</summary>
+        /// <param name="rawText">The raw text to be converted</param>
+        /// <param name="stringBuilder">The string builder which receives the display string</param>
+        protected virtual void MakeDisplayString(string rawText, StringBuilder stringBuilder)
+        {
+            foreach (var c in rawText)
             {
-                buffer.Dequeue();
+                switch (c)
+                {
+                    // Carriage returns are presumed to be followed by line feeds, skip them entirely
+                    case '\r':
+                        continue;
+                    // Newlines become space so that things like multi-line JSON or matricies are still visible
+                    case '\n':
+                    case '\x0085': // Next line character (NEL)
+                    case '\x2028': // Unicode line separator
+                    case '\x2029': // Unicode paragraph separator
+                        stringBuilder.Append(' ');
+                        break;
+                    case '\t':
+                        stringBuilder.Append(c);
+                        break;
+                    // Replace all other control characters with the "�" replacement character
+                    case < ' ': // C0 control characters
+                    case '\x007F': // Delete
+                    case >= '\x0080' and <= '\x009F': // C1 control characters
+                        stringBuilder.Append('\xFFFD');
+                        break;
+                    default:
+                        stringBuilder.Append(c);
+                        break;
+                }
             }
         }
 
+        private void DisplayString(string value)
```

Comment:
```suggestion
        private void AppendString(string value)
```

This is just a nitpick since it is a private method, but when reading the code I was being thrown off by the name since the method doesn't actually display the string.